api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -140,6 +141,31 @@ public StructType partitionType() {
     return lazyPartitionType;
   }
 
+  public StructType originalPartitionType() {
```

Comment:
So would the java doc for this be,
"Returns a struct with TransformID's which match the ID's used in the Table Metadata. The partitionType() method (link) returns a struct which is guaranteed not to overlap with column ID's of the table but this reassigns the ID's from the original table metadata." 

Or something like that?